v1.1: add `csvDelimiter` to add/update/replace document endpoints (#2223)
meilisearch/documentation: reference/api/documents.md

| Query Parameter  | Default Value | Description                                                                   |
| :--------------- | :------------ | :---------------------------------------------------------------------------- |
| **`primaryKey`** | `null`        | [Primary key](/learn/core_concepts/primary_key.md#primary-key-2) of the index |
| **`csvDelimiter`** | `","`        | Configure the character separating CSV fields. Must be a string containing [one ASCII character](https://www.rfc-editor.org/rfc/rfc20). |

If you want to [set the primary key of your index on document addition](/learn/core_concepts/primary_key.md#setting-the-primary-key-on-document-addition), it can only be done **the first time you add documents** to the index. After this, the `primaryKey` parameter will be ignored if given.

> Configuring `csvDelimiter` and sending data in a format other than CSV will cause Meilisearch to throw an error.

Reviewer: Don’t know if it’s worth documenting it, but technically we will not even read the data sent by the user. We will only read the content type.

```suggestion
Configuring `csvDelimiter` and sending data with a content type other than CSV will cause Meilisearch to throw an error.
```

> Configuring `csvDelimiter` and sending data with a content type other than CSV will cause Meilisearch to throw an error.

Author: Yeah, that was my reasoning too, in the end. How strongly do you feel about placing it in the container? I'm a bit against it, but if you think it's crucial I'll do it.

> Configuring `csvDelimiter` and sending data with a content type other than CSV will cause Meilisearch to throw an error.

Reviewer 2: I place information like this in a container. I added [this rule? ](https://www.notion.so/meilisearch/Style-guide-3b07f6d396444b9bbc5dcd52205f3de7?pvs=4#4f650bbb2df64e548c088988838fcc5e)to the style guide after discussing this on Slack a while ago. We can revise it if needed, but for consistency, let's move it to a container for now.

> ### Body

Author: I don't think it's necessary. There are no specific gotchas about `csvDelimiter`, so we/users benefit more from a illustration of the most common scenario, IMHO.

| Query Parameter  | Default Value | Description                                                                       |
| :--------------- | :------------ | :-------------------------------------------------------------------------------- |
| **`primaryKey`** | `null`        | [Primary key](/learn/core_concepts/primary_key.md#primary-key-2) of the documents |
| **`csvDelimiter`** | `","`        | Configure the character separating CSV fields. Must be a string containing [one ASCII character](https://www.rfc-editor.org/rfc/rfc20). |

> Configuring `csvDelimiter` and sending data in a format other than CSV will cause Meilisearch to throw an error.

Reviewer: Same
